### PR TITLE
feat: optimize sdk

### PR DESCRIPTION
**Pull request by zobeirhamid**

### CODE REVIEW

File: packages/sdk/src/BaseAccountAPI.ts [hunk 1]
```diff
@@ -346,9 +345,9 @@ export abstract class BaseAccountAPI {
       initCode,
       callData,
       callGasLimit,
-      verificationGasLimit,
-      maxFeePerGas,
-      maxPriorityFeePerGas,
+      verificationGasLimit: BigNumber.from(await verificationGasLimit).add(await initGas),
```
Comment (derekchiang) [inline packages/sdk/src/BaseAccountAPI.ts:348]: i may be wrong but don't you need `Promise.all` to actually `await` in parallel?  if you do this aren't you still awaiting one by one?

File: packages/sdk/src/Provider.ts [hunk 2]
```diff
@@ -30,10 +28,7 @@ export async function wrapProvider(
   const httpRpcClient = new HttpRpcClient(config.bundlerUrl, config.entryPointAddress, chainId, config.projectId, options?.skipFetchSetup)
 
   const accountAPI = BaseAccountAPI.create(config.implementation.accountAPIClass as unknown as AccountAPIConstructor<any, {}>, {
-    // Use our own provider because some providers like Magic doesn't support custom errors, which
-    // we rely on for getting counterfactual address
-    // Unless it's hardhat.
-    provider: chainId === 31337 ? originalProvider : new ethers.providers.JsonRpcProvider({url: getRpcUrl(chainId), skipFetchSetup: options?.skipFetchSetup ?? undefined}),
+    provider: originalProvider,
```
Comment (derekchiang) [inline packages/sdk/src/Provider.ts:31]: im a little confused about the provider changes -- it doesn't break tests/magic?

Comment (zobeirhamid) [reply packages/sdk/src/Provider.ts:31]: It does not break tests. The key here is when using JsonRpcProvider for Infura it does unnecessary calls where as using Infura directly makes optimzied calls.

File: packages/sdk/src/BaseAccountAPI.ts (hunk 1 above)
Comment (zobeirhamid) [reply packages/sdk/src/BaseAccountAPI.ts:348]: Very valid point, but `verificationGasLimit` and `initGas` limit are called without blocking each other, that's the key optimization here. Using `Promise.all` would not optimize, as while `await verificationGasLimit` `initGas` is still running and most likely will be resolved by the time we call `await initGas`.

**Pull request merged**